### PR TITLE
Add view_index_metadata over metricbeat-* for monitoring agent

### LINKED ISSUE #63203: Metricbeat errors due to missing `admin/aliases/get` permission for `remote_monitoring_user` user
The default metricbeat configuration using the built-in `remote_monitoring_user` user fails due to a missing permission.

The workaround is to create a separate user with the appropriate permissions but the expectation should be that using the built-in users should "just work". My understand would be that either a) metricbeat is doing something it shouldn't or b) the built-in user should have this permission added.

```
Connection marked as failed because the onConnect callback failed: failed to check for alias 'metricbeat-7.9.1': (status=403) {"error":{"root_cause":[{"type":"security_exception","reason":"action [indices:admin/aliases/get] is unauthorized for user [remote_monitoring_user]"}],"type":"security_exception","reason":"action [indices:admin/aliases/get] is unauthorized for user [remote_monitoring_user]"},"status":403}: 403 Forbidden: {"error":{"root_cause":[{"type":"security_exception","reason":"action [indices:admin/aliases/get] is unauthorized for user [remote_monitoring_user]"}],"type":"security_exception","reason":"action [indices:admin/aliases/get] is unauthorized for user [remote_monitoring_user]"},"status":403}
```

### PR DESCRIPTION
**Pull request by albertzaharovits**
The `remote_monitoring_agent` reserved role is extended to grant more privileges over the `metricbeat-*` index pattern.
In addition to the `index` and `create_index` index privileges that it granted already, it now also grants the `view_index_metadata` privilege.
See https://github.com/elastic/elasticsearch/issues/63203 for more details.

The `remote_monitoring_collector` role has not been changed.

Reviewers might prefer to look at the https://github.com/elastic/elasticsearch/commit/ef1eb01178cd1d2754eec6b81a77362962eca975 commit for the diff, as the other commit https://github.com/elastic/elasticsearch/commit/c89d0e901a6023405eb42cb6ea130903bfc89b97 is a rename that clutters a bit the diff of the whole PR.

Closes https://github.com/elastic/elasticsearch/issues/63203

### CODE REVIEW

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java [hunk 1]
```diff
@@ -750,108 +760,116 @@ public void testRemoteMonitoringCollectorRole() {
...
-        assertThat(remoteMonitoringAgentRole.cluster().check(DeleteIndexTemplateAction.NAME, request, authentication), is(false));
-        assertThat(remoteMonitoringAgentRole.cluster().check(ClusterRerouteAction.NAME, request, authentication), is(false));
-        assertThat(remoteMonitoringAgentRole.cluster().check(ClusterUpdateSettingsAction.NAME, request, authentication), is(false));
-        assertThat(remoteMonitoringAgentRole.cluster().check(MonitoringBulkAction.NAME, request, authentication), is(false));
-        assertThat(remoteMonitoringAgentRole.cluster().check(DelegatePkiAuthenticationAction.NAME, request, authentication), is(false));
-
-        assertThat(remoteMonitoringAgentRole.runAs().check(randomAlphaOfLengthBetween(1, 12)), is(false));
-
-        assertThat(remoteMonitoringAgentRole.indices().allowedIndicesMatcher(RecoveryAction.NAME)
+        Role remoteMonitoringCollectorRole = Role.builder(roleDescriptor, null).build();
```
Comment (BigPandaToo) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java:763]: Why did you change it ti remoteMonitoringCollectorRole while the changes are added to remote_monitoring_agent?

Comment (albertzaharovits) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java:763]: It's mainly a rename see: https://github.com/elastic/elasticsearch/commit/c89d0e901a6023405eb42cb6ea130903bfc89b97 .
Besides the rename, because the roles are similar sounding but different in scopee I've tested that only one of them grants the desired privileges.

**Comment by BigPandaToo** [comment]
LGTM

**Review by BigPandaToo — APPROVED** [review]

**Comment by albertzaharovits** [comment]
Thank you @BigPandaToo !

**Comment by albertzaharovits** [comment]
It will be in 7.10 BC3.
